core/plugins/org.fusesource.ide.camel.model.service.impl/src/org/fusesource/ide/camel/model/service/internal/CamelService.java

```diff
@@ -166,10 +169,23 @@ public CamelModel getCamelModelForKarafRuntimeProvider(String camelVersion) {
 	 */
 	@Override
 	public CamelModel getCamelModel(String camelVersion, String runtimeProvider) {
+		return getCamelModel(camelVersion, runtimeProvider, new NullProgressMonitor());
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.fusesource.ide.camel.model.service.core.ICamelManagerService#getCamelModel(java.lang.String, java.lang.String, org.eclipse.core.runtime.IProgressMonitor)
+	 */
+	@Override
+	public CamelModel getCamelModel(String camelVersion, String runtimeProvider, IProgressMonitor monitor) {
 		CamelCatalogCoordinates coords = CamelCatalogUtils.getCatalogCoordinatesFor(runtimeProvider, camelVersion);
+		SubMonitor subMonitor = SubMonitor.convert(monitor, "Downloading Camel Dependencies for version " + coords.getVersion(), 4);
```

Comment:
That will be problematric later as you have a PR ready which will introduce even more impl bundles and all of these then must duplicate that externalization.

Reply:
Don't understand why it would be problematic, requires more work yes.
Anyway, I think that it is not a part of the duplicated code, see https://github.com/jbosstools/jbosstools-fuse/pull/1337/files#diff-6f3995e9cde8ad715b8ba14e9d12872aL171